### PR TITLE
Clean up the code, add docstrings, and get rid of the poorly implemented async client

### PR DESCRIPTION
**Pull request by wojcikstefan**
Right now the async client makes the code more complex and doesn't really add much value. I doubt it's used in the wild and if it is, then the usage is probably quite confusing and hacky. I think we might as well get rid of it.

Examples of current confusing behavor:
* `api.get/post/put/delete` is still synchronous
* `api.map` expects a list of `grequests.AsyncRequest`, but there's no easy way to construct them.

This PR introduces breaking changes and should be published along with a major version bump.

TODO:
- [x] Add mocked unit tests
- [ ] Bump version to v1.0
- [ ] Add a GitHub Release and publish on PyPI

### CODE REVIEW

**Comment by wojcikstefan** [comment]
Bonus point against async code - due to monkey patching upon import, grequests can cause warnings/bugs that might be very hard to find. See:
* https://github.com/kennethreitz/grequests/issues/100
* https://github.com/kennethreitz/grequests/issues/99
* https://github.com/kennethreitz/grequests/issues/97
* https://github.com/kennethreitz/grequests/issues/88
* https://github.com/kennethreitz/grequests/issues/86

**Comment by thomasst** [comment]
Looks good (but didn't test).

**Comment by wojcikstefan** [comment]
@closeio/engineering I believe that once we add unit tests to this PR, we can release it as v1.0. Agreed?

**Comment by philfreo** [comment]
👍

**Comment by thomasst** [comment]
Sounds good. So what's happening to async? Are there any plans for it?

**Comment by wojcikstefan** [comment]
Given the arguments I laid out in https://github.com/closeio/closeio-api/pull/78#issuecomment-275871159 and in the description for #79, I think it might make sense to drop it altogether. This way we maintain a simple & readable helper library w/o any quirks while the users can choose to run multiple concurrent `api.get/post/put/delete` requests via `threading`, `multiprocessing`, `concurrent.futures`, `gevent`, etc. depending on their specific use case and environment.

File: closeio_api/__init__.py [hunk 1]
```diff
@@ -23,136 +27,137 @@ def __init__(self, response):
...
+                 verify=True):
         assert base_url
         self.base_url = base_url
-        self.async = async
         self.max_retries = max_retries
         self.tz_offset = tz_offset or str(local_tz_offset())
         self.verify = verify
 
-        if async:
-            import grequests
```
Comment (philfreo) [inline closeio_api/__init__.py]: You can probably remove grequests etc now from setup.py and requirements.txt

Comment (wojcikstefan) [inline closeio_api/__init__.py]: Thanks, done in 	02bad5b

**Comment by philfreo** [comment]
@wojcikstefan Let's move forward with merging this? Can you handle conflicts and publish?

**Comment by wojcikstefan** [comment]
@thomasst @jkemp101 @philfreo This should be ready to go, please review. If you give the go ahead, I'll merge the changes, bump up the version, create a new release, and deploy to PyPI.

**Pull request merged**